desc/magnetic_fields.py

```diff
@@ -1094,3 +1129,614 @@ def odefun(rpz, s):
+            the form potential(theta,zeta,**params) -> ndarray.
+            theta,zeta are poloidal and toroidal angles on the surface
+        potential_dtheta: callable
+            function to compute the theta derivative of the current potential
+        potential_dzeta: callable
+            function to compute the zeta derivative of the current potential
+        params : dict, optional
+            default parameters to pass to potential function (and its derivatives)
+
+        """
+        errorif(
+            not isinstance(surface, FourierRZToroidalSurface),
+            TypeError,
+            "Expected type FourierRZToroidalSurface for argument surface, "
+            f"instead got type {type(surface)}",
+        )
+
+        R_lmn = surface.R_lmn
+        Z_lmn = surface.Z_lmn
+        modes_R = surface._R_basis.modes[:, 1:]
+        modes_Z = surface._Z_basis.modes[:, 1:]
+        NFP = surface.NFP
+        sym = surface.sym
+        rho = surface.rho
+        name = surface.name
+
+        return cls(
+            potential,
+            potential_dtheta,
+            potential_dzeta,
+            params,
+            R_lmn,
+            Z_lmn,
+            modes_R,
+            modes_Z,
+            NFP,
+            sym,
+            rho,
+            name,
+            check_orientation=False,
+        )
+
+
+class FourierCurrentPotentialField(_MagneticField, FourierRZToroidalSurface):
+    """Magnetic field due to a surface current potential on a toroidal surface.
+
+        surface current K is assumed given by
+
+        K = n x ∇ Φ
+        Φ(θ,ζ) = Φₛᵥ(θ,ζ) + Gζ/2π + Iθ/2π
+
+        where:
+              n is the winding surface unit normal.
+              Phi is the current potential function,
+                which is a function of theta and zeta,
+                and is given as a secular linear term in theta/zeta
+                and a double Fourier series in theta/zeta.
+        This function then uses biot-savart to find the
+        B field from this current density K on the surface.
+
+    Parameters
+    ----------
+    Phi_mn : ndarray
+        Fourier coefficients of the double FourierSeries part of the current potential.
+    modes_Phi : array-like, shape(k,2)
+        Poloidal and Toroidal modenumbers corresponding to passed-in Phi_mn coefficients
+    I : float
+        Net current linking the plasma and the surface toroidally
+        Denoted I in the algorithm
+    G : float
+        Net current linking the plasma and the surface poloidally
+        Denoted G in the algorithm
+        NOTE: a negative G will tend to produce a positive toroidal magnetic field
+        B in DESC, as in DESC the poloidal angle is taken to be positive
+        and increasing when going in the clockwise direction, which with the
+        convention n x grad(phi) will result in a toroidal field in the negative
+        toroidal direction.
+    sym_Phi : str or False, one of {"auto","cos","sin",False}
```

Comment:
when the options are a discrete set like this you don't have to give the type, ie can just write `sym_Phi : {"auto", "cos", "sin", False}` with the default option first